WordPress/Classes/Networking/ServiceRemoteREST.h

```diff
@@ -28,6 +34,15 @@ typedef enum
  */
 - (id)initWithApi:(WordPressComApi *)api;
 
+/**
+ *  @brief      Designated initializer.
+ *
+ *  @param      api     The API to use for communication.  Cannot be nil.
+ *
+ *  @returns    The initialized object.
+ */
+- (id)initWithComRestApi:(WordPressComRestApi *)api;
```

Comment:
Even then, I'd feel much safer transitioning with a separate class. The main problem is that you can now initialize this with a `WordPressComApi` or a `WordPressComRestApi`, and so the class can be two different things.